### PR TITLE
Remove propagation.utils

### PR DESCRIPTION
**Pull request by ZStriker19**
## Commit Message
Moving propagation.utils to trace_utils

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: releasenotes/notes/move_propagation.utils_to_trace_utils-6bdfed9fcb98a995.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    The ``propagation.utils`` module has been removed. Public methods have been moved to ``trace_utils.py``
```
Comment (brettlangdon) [inline releasenotes/notes/move_propagation.utils_to_trace_utils-6bdfed9fcb98a995.yaml:4]:
```suggestion
    The ``propagation.utils`` module has been removed. Public methods have been moved to ``ddtrace.contrib.trace_utils``
```

Comment (brettlangdon) [reply releasenotes/notes/move_propagation.utils_to_trace_utils-6bdfed9fcb98a995.yaml:4]: we just want to state the module path that they were moved to, the file name won't help anyone.

**Review by brettlangdon — APPROVED** [review]

**Review by brettlangdon — CHANGES_REQUESTED** [review]
a bunch of tests are failing

**Comment by ZStriker19** [comment]
Waiting on this PR for the moment because @Kyle-Verhoog  brought up that perhaps we should just deprecate all of propagation and create a base class for propagation for users to implement.

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]
We should do the deprecation PR first (targeting master) then update the 1.0-dev branch with master and then do this PR with the removal

**Comment by ZStriker19** [comment]
Syncing with Kyle: Do we need to actually re-expose these helpers publicly through `trace.utils`?

**Comment by majorgreys** [comment]
Moved to https://github.com/DataDog/dd-trace-py/pull/3236

**Pull request closed without merging**